File: src/main/java/com/kingdom/repository/UserRepository.java
package com.kingdom.repository;

import com.kingdom.model.User;
import org.springframework.data.repository.CrudRepository;

import java.util.List;

public interface UserRepository extends CrudRepository<User, Integer> {

    List<User> findAllByOrderByLastLogin();

    User findByEmail(String email);

    User findByUsername(String username);

    User findByUsernameAndPassword(String username, String password);
}

File: src/main/java/com/kingdom/repository/UserRepository.kt
package com.kingdom.repository

import com.kingdom.model.User
import org.springframework.data.repository.CrudRepository

interface UserRepository : CrudRepository<User, Int> {

    fun findAllByOrderByLastLogin(): List<User>

    fun findByEmail(email: String): User

    fun findByUsername(username: String): User

    fun findByUsernameAndPassword(username: String, password: String): User
}
